app/models/user.rb

```diff
@@ -30,8 +30,30 @@ def event_manager?
     !manager_profile.nil?
   end
 
+  def self.user_info(parameter)
+    if parameter[:user_id].nil?
+      {error: "Field is empty"}
+    else
+      user_role = user_role(parameter[:role])
+      user_data = user_hash(parameter).merge(user_role: user_role)
+      parameter.merge(user_data).symbolize_keys
+    end
+  end
+
   def generate_auth_token
     payload = { user_id: id, email: email }
     AuthToken.encode(payload)
   end
+
+  private
+
+  def self.user_role(role)
```

Comment:
`private` (on line 48) does not make singleton methods private. Use `private_class_method` or `private` inside a `class << self` block instead.